### PR TITLE
Added support for resource-specific destination table for Firewall

### PR DESCRIPTION
**Pull request by picccard**
## Overview/Summary

This allows consumers to use the resource-specific destination table in diag settings for AzFirewalls. Ref. issue #1283

## This PR fixes/adds/changes/removes

1. Adds support for setting resource-specific destination table in diagnostic setting for Azure Firewall

### Breaking Changes

None, optional parameter. Default value is the same as todays default value.

## Testing Evidence

Please provide any testing evidence to show that your Pull Request works/fixes as described and planned (include screenshots, if appropriate).

#### Azure Public

[Deploy To Azure](https://portal.azure.com/#blade/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fpicccard%2FEnterprise-Scale%2Fdiagsettings-afw-destinationtype%2FeslzArm%2FeslzArm.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2Fpicccard%2FEnterprise-Scale%2Fdiagsettings-afw-destinationtype%2FeslzArm%2Feslz-portal.json)

#### Azure US Gov (Fairfax)
[Deploy To Azure](https://portal.azure.us/#blade/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fpicccard%2FEnterprise-Scale%2Fpicccard%2FeslzArm%2FeslzArm.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2Fpicccard%2FEnterprise-Scale%2Fdiagsettings-afw-destinationtype%2FeslzArm%2Ffairfaxeslz-portal.json)

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Enterprise-Scale/pulls)
- [x] Associated it with relevant [issues](https://github.com/Azure/Enterprise-Scale/issues), for tracking and closure.
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Enterprise-Scale/tree/main)
- [ ] Performed testing and provided evidence.
- [ ] Ensured [contribution guidance](https://github.com/Azure/Enterprise-Scale/wiki/ALZ-Contribution-Guide) is followed.
- [x] Updated relevant and associated documentation.
- [x] Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located: `/docs/wiki/whats-new.md`)

### CODE REVIEW

**Review by Springstone — APPROVED** [review]
LGTM

**Comment by jtracey93** [comment]
@picccard & @Springstone seems there is an issue with the template here, can you investigate https://github.com/Azure/Enterprise-Scale/actions/runs/5120770157/jobs/9207677548?pr=1334

File: src/resources/Microsoft.Authorization/policyDefinitions/Deploy-Diagnostics-Firewall.json [hunk 1]
```diff
@@ -138,6 +153,7 @@
                     "dependsOn": [],
                     "properties": {
                       "workspaceId": "[[parameters('logAnalytics')]",
+                      "logAnalyticsDestinationType": "[[parameters('logAnalyticsDestinationType')]"
```
Comment (Springstone) [inline src/resources/Microsoft.Authorization/policyDefinitions/Deploy-Diagnostics-Firewall.json:156]: This is the issue, there is a "," missing at the end of the line.

**Review by Springstone — CHANGES_REQUESTED** [review]
Found the issue.

**Review by jtracey93 — APPROVED** [review]

**Pull request merged**